doc/src/guards_in_macro.md

```diff
@@ -0,0 +1,70 @@
+# Guards in macro
+
+Sometimes, you may want two very similar set of declarations, but with a few changes. A typical example is when implementing a Rust-like language with optional trailing ";" in some control-flow expressions within blocks. In that case, you want to allow omitting the trailing ";" only after a block, because in that case the grammar is unambiguous.
```

Comment:
I feel like the start of this is a bit vague/couched, I was having a hard time understanding what the setup + use case is.

Can we use an example here? I think we want to say something like "you want to do this but Lalrpop tells you your grammar is ambigious. We know some additional contextual information that we can pass through the rules which will be expanded into two slightly different cases. Lalrpop now accepts this grammar and the input as expected while rejecting this other thing we didn't want to accept `{1 2}`.

If you have a different way to show this off with examples go for it!